lambda_functions/execute_ssh_command_js/README.md

```diff
@@ -1,13 +1,72 @@
-
-
 ## Setup
 
 Since this plugin uses the Serverless plugin `serverless-secrets-plugin` you need to setup the `node_modules` by running:
 
-    npm install
+    1) Follow nvm installation guide as mentioned in https://github.com/creationix/nvm. 
+    2) cd to the folder where nvm.sh is installed and run the following command to source to ~/.bashrc, or ~/.profile, or ~/.zshrc file:
+          `. nvm.sh`
+    3) Run the following shell commands (to check if installation is completed):
+        a) `nvm --version`
+        b) `nvm install node`
+        c) `nvm use node`
+        d) `node --version`
+        e) `nvm install npm`
+        f) `npm --version`
+        g) `npm install -g serverless`
+        h) `serverless --version`
+        i) `serverless config credentials --provider aws --key <AWS key> --secret <AWS Secret key>`
+
+## Installation before deployment
+
+    1) Run the following shell commands (to install dependencies):
+        a) `npm install serverless-pseudo-parameters --save-dev`
+        b) `npm install simple-ssh --save-dev`
+        c) `serverless --version`
+    2) Ensure that ssh private and public keys are generated on the Raijin system.
+    3) Private key is copied to aws ssm parameter using the following aws command line:
+           aws ssm put-parameter --name "orchestrator.raijin.users.default.pkey" --type "SecureString" --value "-----BEGIN RSA PRIVATE KEY-----
+            MIIJJgIBAAKCAgEA3lkvu08KVjA7hWyFnKo+0Eb/S1SCZxIgwlfDAjlJexdQdh1y
+            ..........
+            MT7OOmFZxUVThbi1Hl6VLzA+cIImOVVrfvtfDlgvG9mNWVvONIVDeIma
+            -----END RSA PRIVATE KEY-----" --overwrite
```

Comment:
The `aws` command includes a very useful feature for [Loading Parameters from a File](https://docs.aws.amazon.com/cli/latest/userguide/cli-using-param.html#cli-using-param-file).

We should be able to say `aws ssm put-parameter ... --value file://~/.ssh/my_private_key_file --overwrite`

Again, this is more secure than pasting on the command line.